code/modules/clothing/head/_head.dm

```diff
@@ -27,4 +29,25 @@
 	..()
 	if(ismob(loc))
 		var/mob/M = loc
-		M.update_inv_head()
\ No newline at end of file
+		M.update_inv_head()
+
+/obj/item/clothing/head/throw_impact(atom/hit_atom, datum/thrownthing/thrownthing)
+	. = ..()
+	if(!.)
+		return
+	if(thrownthing && thrownthing.target_zone && thrownthing.target_zone != BODY_ZONE_HEAD)
+		return
+	if(iscarbon(hit_atom))
+		var/mob/living/carbon/H = hit_atom
+		if(H.head && istype(H.head, /obj/item/clothing/head))

```

Comment:
can remove the first H.head here, istype handles nulls

Reply:
Isn't this redundant due to the other istype at line 44?

Reply:
Someone give a definitive answer for this please?

Reply:
That depends if you want the bounce off message, if they're wearing non-clothing/head.